packages/dev/core/src/Rendering/boundingBoxRenderer.ts

```diff
@@ -483,6 +533,229 @@ export class BoundingBoxRenderer implements ISceneComponent {
         engine.currentRenderPassId = currentRenderPassId;
     }
 
+    /**
+     * Sets whether to use instanced rendering.
+     * When not enabled, BoundingBoxRenderer renders in a loop,
+     * calling engine.drawElementsType for each bounding box in renderList,
+     * making every bounding box 1 or 2 draw call.
+     * When enabled, it collects bounding boxes to render,
+     * and render all boxes in 1 or 2 draw call.
+     * This could make the rendering with many bounding boxes much faster than not enabled,
+     * but could result in a difference in rendering result if
+     * {@link BoundingBoxRenderer.showBackLines} enabled,
+     * because drawing the black/white part of each box one after the other
+     * can be different from drawing the black part of all boxes and then the white part.
+     * Also, when enabled, events of {@link BoundingBoxRenderer.onBeforeBoxRenderingObservable}
+     * and {@link BoundingBoxRenderer.onAfterBoxRenderingObservable} would only be triggered once
+     * for one rendering, instead of once every bounding box.
+     * Events would be triggered with a dummy box to keep backwards compatibility,
+     * the passed bounding box has no meaning and should be ignored.
+     * @param val whether to use instanced rendering
+     */
+    set useInstances(val: boolean) {
+        this._useInstances = val;
+        if (this._colorShader) {
+            this._colorShader.setDefine("INSTANCES", val);
+        }
+        if (!val) {
+            this._cleanupInstances();
+        }
+    }
+
+    get useInstances(): boolean {
+        return this._useInstances;
+    }
+
+    /**
+     * Instanced render the bounding boxes of a specific rendering group
+     * @internal
+     * @param renderingGroupId defines the rendering group to render
+     */
+    public _renderInstanced(renderingGroupId: number): void {

```

Comment:
Should be `protected` or `private`. Also, you don't have to provide documentation for protected/private methods (but you can if you want!).